oak-run/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/LuceneIndexer.java

```diff
@@ -111,11 +114,22 @@ private void writeToIndex(Document doc, String path) throws IOException {
     private LuceneDocumentMaker newDocumentMaker(IndexDefinition.IndexingRule indexingRule, String path) {
         return new LuceneDocumentMaker(
                 binaryTextExtractor,
-                () -> FacetHelper.getFacetsConfig(definitionBuilder), //TODO FacetsConfig handling
-                null,   //TODO augmentorFactory
+                // we re-use the facet config
+                this,
+                // augmentorFactory is not supported (it is deprecated)
+                null,
                 definition,
                 indexingRule,
                 path
         );
     }
+
+    @Override
+    public FacetsConfig getFacetsConfig() {
+        if (facetsConfig == null){
+            facetsConfig = FacetHelper.getFacetsConfig(definitionBuilder);
```

Comment:
Would this be a good thing to log so the user can diagnose?

Reply:
I don't know if it's really needed... the old code just seems completely broken by design, and the TODO even admits that... it's just, the TODO was never fixed, and there is a complete lack of tests in this area. I would rather fix that instead of adding a way to analyze... just my opinion.